secrets/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// return codes:
+// 1: failed to read secret from env
+// 2: failed to parse hierarchical secret
+// 3: failed to write secret content into file
+func main() {
+	for _, name := range os.Args[1:] {
+		i := strings.Index(name, ":")
+		var keys []string
+		if i > 0 {
+			keys = strings.Split(name[i+1:], ",")
+			name = name[:i]
+		}
+		value, ok := os.LookupEnv(name)
+		if !ok {
+			fmt.Fprintf(os.Stderr, "%q variable not set", name)
+			os.Exit(1)
+		}
+
+		secrets := filepath.Join("/run/secrets", name)
```

Comment:
We probably need to sanitise `name` to prevent path traversal (`name=../../etc/passwd`)

Reply:
this runs from a scratch container, so would have no impact